src/librustc_mir/interpret/const_eval.rs

```diff
@@ -98,44 +98,20 @@ pub fn value_to_const_value<'a, 'tcx>(
     val: Value,
     ty: Ty<'tcx>,
 ) -> &'tcx ty::Const<'tcx> {
-    let layout = tcx.layout_of(ty::ParamEnv::reveal_all().and(ty)).unwrap();
-
-    if layout.is_zst() {
-        return ty::Const::from_const_value(
-            tcx,
-            ConstValue::ByVal(PrimVal::Undef),
-            ty);
-    }
-
-    let val = match layout.abi {
-        layout::Abi::Scalar(..) => {
-            if let Value::ByVal(val) = val {
-                ConstValue::ByVal(val)
-            } else {
-                bug!("expected ByVal value, got {:?}", val);
-            }
-        }
-        layout::Abi::ScalarPair(..) => {
-            if let Value::ByValPair(a, b) = val {
-                ConstValue::ByValPair(a, b)
-            } else {
-                bug!("expected ByValPair value, got {:?}", val);
```

Comment:
Please keep these sanity checks. At least when debug-assertions is on. I really don't trust Miri code to be correct.